fiftyone/core/view.py

```diff
@@ -238,7 +244,10 @@ def default_group_slice(self):
 
     @property
     def name(self):
-        """The name of the view."""
+        """The name of the view with which it was saved. Defaults to `<dataset_name>-view` for unsaved views."""
```

Comment:
nit: unfortunately the linter doesn't automatically enforce line length limits for comments, but our style is to always keep everything to 79 characters or fewer.

My two cents would be to document like this:

```py
    @property
    def name(self):
        """The name of the view.
    
        If this view is a saved view, its saved name will be returned.
        Otherwise, a default name will be returned.
        """
```

Reply:
ah didn't notice the line length didn't auto-format correctly. I'll see if there's a setting in my IDE to do it automatially.

Reply:
On second thought, thinking about the saved view selection dropdown impl/the usage in the App, does it make sense to return the `<dataset_name>-view` as the default? Maybe it would make life easier to make the default something along the lines of `Unsaved`. What do you think?

Reply:
It sounds like you're thinking of `DatasetView.name` as a way to communicate to the App whether the current view is saved. For that purpose, I would recommend using the `is_saved` pattern I mentioned:

```py
dataset_name = view._dataset.name
stages = view._serialize()
saved_name = view.name if view.is_saved else None  # use None for unsaved here
```

For the SDK, `SampleCollection.name` is declared mostly for cosmetic reasons. Here's one place where its used currently:

```py
import fiftyone as fo
import fiftyone.zoo as foz

dataset = foz.load_zoo_dataset("quickstart", max_samples=1)
view = dataset.select_fields()

d = view.to_dict()

fo.pprint(d)

"""
{
    'name': 'quickstart-1-view',
    'version': '0.18.0',
    'media_type': 'image',
    ...
}
"""

dataset2 = fo.Dataset.from_dict(d)
print(dataset2.name)  # quickstart-1-view
```

It's not really essential that the exported JSON contains the source dataset name as root, but my thinking before saved views existed was just to leave a breadcrumb in the JSON back to the original dataset.

Reply:
However, now that saved views exist, perhaps we should just return `name == None` for unsaved views? That would be the most clear way to indicate to the user that views *can* have well-defined names, but this one doesn't have one.

I don't think any existing usages of `DatasetView.name` would mind if `name == None`.

Reply:
If we do go with this change, then I would be fine with getting rid of `DatasetView.is_saved` and just having the convention that unsaved views have `name=None`.

Reply:
FYI I pushed a commit that fixes the only potential problem I saw if we allow `DatasetView.name=None`